### PR TITLE
balancerd: Align layout with `clusterd` and `environmentd`

### PR DESCRIPTION
**Pull request by ParkMyCar**
This PR is pure code movement, it aligns the layout of `balancerd` to match that of `clusterd` and `environmentd`. Folks should feel free to push back if they don't think it's necessary, this is just something I noticed while working in the area.

1. Rename `main.rs` to `balancerd.rs` and merge `service.rs` into `balancerd.rs`
2. Move the mzbuild setup from `misc/images/balancerd` to `src/balancerd/ci`

### Motivation

* This PR refactors existing code.

Aligns the filesystem layout of `balancerd` with `clusterd` and `environmentd`

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

**Comment by sdht0** [comment]
While at it:
* Also move `src/environmentd/src/bin/environmentd/main.rs` to `src/environmentd/src/bin/environmentd/environmentd.rs`?
* Maybe out of scope, but should we rename the crate name of `persist-cli` from the current `name = "persistcli"` to `name = "persist-cli"` for consistency? Discovered when [exploring crate intra-dependencies](https://www.notion.so/materialize/Materialize-intra-crate-dependencies-d0c43f9c2698435781e05cf5db5a884b).

**Comment by ParkMyCar** [comment]
> * Maybe out of scope, but should we rename the crate name of `persist-cli` from the current `name = "persistcli"` to `name = "mz-persist-cli"` for consistency? Discovered when [exploring crate intra-dependencies](https://www.notion.so/materialize/Materialize-intra-crate-dependencies-d0c43f9c2698435781e05cf5db5a884b).

Good question! Let me check with Persist folks, will post in the team channel

File: src/balancerd/src/lib.rs [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 mod codec;
 mod dyncfgs;
+// mod service;
```
Comment (jkosh44) [inline src/balancerd/src/lib.rs:21]: I think you may have left this in by mistake.

**Review by jkosh44 — APPROVED** [review]
LGTM

**Review by def- — APPROVED** [review]
No complaints from QA side

File: src/balancerd/src/lib.rs (hunk 1 above)
Comment (ParkMyCar) [reply src/balancerd/src/lib.rs:21]: I totally did, thanks!

**Review by jubrad — APPROVED** [review]

**Pull request merged**